src/state/trajectory/logics.ts

```diff
@@ -345,6 +353,102 @@ const loadFileViaUrl = createLogic({
     type: LOAD_FILE_VIA_URL,
 });
 
+// configures the controller for file conversion and sends server health checks
+const configureFileConversionLogic = createLogic({
+    process(
+        deps: ReduxLogicDeps,
+        dispatch: <T extends AnyAction>(action: T) => T,
+        done
+    ) {
+        const { getState } = deps;
+
+        // TODO: Most likely this will eventually replace the config up top
+        // but for development purposes it's here
+        // until we switch to Octopus, make sure it matches your local instance.
+        const netConnectionConfig: NetConnectionParams = {
+            serverIp: "0.0.0.0",
+            serverPort: 8765,
+            useOctopus: true,
+            secureConnection: false,
+        };
+        // check if a controller exists and has the right configuration
+        let controller = getSimulariumController(getState());
+        if (!controller || !controller.remoteWebsocketClient) {
+            dispatch(
+                setConversionStatus({
+                    status: CONVERSION_NO_SERVER,
+                })
+            );
```

Comment:
why do you do this here? aren't you going to be checking the server below anyway?

Reply:
yes, but there could be a case where the server was previously healthy, and now is not, and the attempt to set the controller below could fail, so it makes sense in the meanwhile to change status to `CONVERSION_NO_SERVER` just in case. that was my thinking at least

Reply:
But isn't the point of this status to show the error modal? If you want it to default to that, you could just have it be the default in the reducer. But here it seems like an unnecessary setting of state

Reply:
It is the default state in the reducer. The case I'm trying o to anticipate is: we have had done a successful check, and set state to `CONVERSION_SERVER_LIVE` but then something happened to the network or the controller in the meanwhile. I'm trying to use `conversionState` as a way to track the whole process, get in front of bugs and and make the user facing interactions like modals easy to time and control.

But I think it's making review for this PR more complicated than it needs to be. I'll just remove the useEffect check on pageload, and this status setting.

If there are bugs in these kind of cases when octopus is actually live we can figure it out then. It will probably work fine this way.

Reply:
yeah, that's overkill. The overall goal is to just check it once, and if it's up assume we're good to go.

Reply:
following from our other discussion yesterday there is still a call on page load, but we've moved things into the action formerly known as `convertFile` and removed the `onclick` function from the load file menu